### PR TITLE
STN-325: Private Pages

### PR DESCRIPTION
**Pull request by kaseybon**
# READY FOR REVIEW

## Summary
This pull request adds the following:
- A lock icon to the left of the title in private pages
- A lock icon to the left of the link text for private pages in the main nav and menu blocks
- A utility class (`hs-private-link`) that adds a lock icon to the left of link text in WYSIWYG editors

Work has been reviewed in the following browsers:
- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge
- [x] IE 11

This has been given a design approval from J. All icons should now be black instead of the primary color so they don't exactly match the screenshots below.

## Need Review By (Date)
6/25

## Urgency
medium

## Steps to Test
1. Run `npm test`
2. Import changes to the ckeditor config files
3. Create a new private page (content -> private page) and add the page to the main navigation
4. - [x] There should be a lock icon next the link the main and secondary navigation regardless of the nesting level (see video in Jira card for more details on what this should look like)
    Private_Page___Department_of_Economics_and_Slack_____-general___Sparkbox
5. - [x] There should be a lock icon next to the title of the private page.
    Private_Page___Department_of_Economics_and_Slack_____-general___Sparkbox
6. - [x] There should be an option to add the `hs-private-link` class to inline links in the ckeditor
    Screen_Shot_2020-06-23_at_1_45_04_PM
    Private_Page___Department_of_Economics

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

File: docroot/themes/humsci/humsci_basic/humsci_basic.theme [hunk 1]
```diff
@@ -107,6 +111,12 @@ function _humsci_basic_check_link_access(array &$link_items) {
       $parameters = $url->getRouteParameters();
       $nid = $parameters['node'];
       $node = Node::load($nid);
+      $type = $node->bundle();
+
+      if ($node && $type == 'hs_private_page') {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/humsci_basic.theme:116]: cool!

File: docroot/themes/humsci/humsci_basic/src/scss/components/_private.scss [hunk 2]
```diff
@@ -0,0 +1,73 @@
+.private-page {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_private.scss]: should we call this file private-page?

File: docroot/themes/humsci/humsci_basic/src/scss/tools/_functions.icons.scss [hunk 3]
```diff
@@ -26,5 +26,7 @@
...
   } @else if ($icon == 'arrow-right-long') {
     @return '<svg viewBox="0 0 22 14"><g id="Page-1" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd"><g id="hb-traditional-arrow" transform="translate(0.000000, 1.000000)" stroke="#{$color}" stroke-width="2"><line x1="0" y1="6" x2="20" y2="6" id="Path"></line><polyline id="Path" points="14 0 20 5.92308 14 11.5"></polyline></g></g></svg>';
+  } @else if ($icon == 'lock') {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/tools/_functions.icons.scss:29]: 👍

File: docroot/themes/humsci/humsci_basic/src/scss/components/_private.scss [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        height: hb-calculate-rems(36px);
+        width: hb-calculate-rems(36px);
+        margin-right: hb-spacing-width('xxxs');
+        margin-top: hb-calculate-rems(-8px);
+      }
+    }
+  }
+}
+
+.hb-private-page-link {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_private.scss:27]: I think this should go in utilities? or no?

File: docroot/themes/humsci/humsci_basic/src/scss/components/_private.scss [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+  .hb-main-nav__menu-lv2 & {
+    @include grid-media-min('lg') {
+      background-size: hb-calculate-rems(15px) hb-calculate-rems(15px);
+      background-position: hb-calculate-rems(7px) hb-calculate-rems(10px);
+    }
+  }
+}
+
+.hs-private-link {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_private.scss]: I think this should go in the utilities file right?

File: docroot/themes/humsci/humsci_basic/src/scss/components/_private.scss [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  @include grid-media-min('md') {
+    background-position: hb-calculate-rems(14px) hb-calculate-rems(24px);
+  }
+
+  @include grid-media-min('lg') {
+    background-position: hb-calculate-rems(14px) hb-calculate-rems(15px);
+  }
+
+  .hb-main-nav & {
+    @include grid-media-min('lg') {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_private.scss]: nice!!

File: docroot/themes/humsci/humsci_basic/src/scss/components/_private.scss [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+      background-position: left 0 bottom hb-calculate-rems(20px);
+      padding-left: hb-calculate-rems(24px) !important;
+    }
+  }
+
+  .hb-main-nav__menu-lv2 & {
+    @include grid-media-min('lg') {
+      background-size: hb-calculate-rems(15px) hb-calculate-rems(15px);
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/components/_private.scss]: also a good edge case catch!

**Review by silverli — CHANGES_REQUESTED** [review]
Hey this looks great! Just one or two questions and then upon reviewing I saw this one tiny potential bug:

Screen Shot 2020-06-26 at 4 45 54 PM

**Comment by kaseybon** [comment]
@silverli since it's an inline pseudo element that can happen at some screen sizes. It's the same thing with the external link icon as well.
Screen_Shot_2020-06-26_at_4_59_49_PM

File: docroot/themes/humsci/humsci_basic/src/scss/utilities/_wysiwyg-text-area.scss [hunk 8]
```diff
@@ -285,6 +285,23 @@ a[href*="//"].hs-external-link,
   @include hb-external-link-icon;
 }
 
+.hs-private-link {
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/src/scss/utilities/_wysiwyg-text-area.scss:288]: 👏

**Review by silverli — APPROVED** [review]
looks great!!

**Pull request merged**